draft-ietf-ppm-dap.md

```diff
@@ -2155,14 +2138,50 @@ structure includes the following:
   reports included in the batch, such that the interval's start and duration are
   both multiples of the task's `time_precision` parameter. Note that in the case
   of a time-interval query (see {{batch-mode}}), this interval can be smaller
-  than the one in the corresponding `CollectionReq.query`.
+  than the one in the corresponding `CollectionJobReq.query`.
 
 * `leader_encrypted_agg_share`: The Leader's aggregate share, encrypted to the
   Collector.
 
 * `helper_encrypted_agg_share`: The Helper's aggregate share, encrypted to the
   Collector.
 
+If the Leader finds the `CollectionJobReq` to be valid, it immediately responds
+with HTTP status 201 Created with a body containing a `CollectionJobResp`. The
+`status` field is set to `processing`, and the response contains a Location
+header containing the relative reference
+`/tasks/{task-id}/collection_jobs/{collection-job-id}`. The Leader SHOULD
```

Comment:
Q. What's the justification for including a Location field, here or in the aggregation interaction? In both cases, the URI included in the Location header field is entirely determined by the specification, and the HTTP client has enough information to construct the URI themselves.

Reply:
The rationale is (unfortunately) subtle.

https://www.rfc-editor.org/rfc/rfc9110#name-201-created

> The 201 (Created) status code indicates that the request has been fulfilled and has resulted in one or more new resources being created. The primary resource created by the request is identified by either a [Location](https://www.rfc-editor.org/rfc/rfc9110#field.location) header field in the response or, if no [Location](https://www.rfc-editor.org/rfc/rfc9110#field.location) header field is received, by the target URI.
>
> The 201 response content typically describes and links to the resource(s) created. Any validator fields ([Section 8.8](https://www.rfc-editor.org/rfc/rfc9110#response.validator)) sent in the response convey the current validators for a new representation created by the request. Note that the PUT method ([Section 9.3.4](https://www.rfc-editor.org/rfc/rfc9110#PUT)) has additional requirements that might preclude sending such validators.

Consider aggregation: we PUT at `{helper}/tasks/{task-id}/aggregation_jobs/{aggregation-job-id}`, BUT we expect the Leader to poll the job at `{helper}/tasks/{task-id}/aggregation_jobs/{aggregation-job-id}?step=0`. That is, the resource we're polling is at a different URI than the one that was provided in the PUT.

However, this is not true for collection. So I think we can (and should) eschew the Location header for collection. I will edit.